### PR TITLE
Fix: Hide the link to work queues for push work pools

### PR DESCRIPTION
**Pull request by zhen0**
Push work pools do not use work queues and the work queues tab is hidden on the work pool page for any push type work pool.  However, we are still showing the work queue as default and displaying a link to a work queue in the deployment and flow run pages (and flow run list item).  This PR adds a check for whether a work pook is a push type work pool and, if it's a push type pool, hides the link to a work queue. 

Current:

image

Updated:

image

### CODE REVIEW

File: src/components/DeploymentDetails.vue [hunk 1]
```diff
@@ -107,6 +107,9 @@
   const api = useWorkspaceApi()
   const updateScheduleLoading = ref(false)
 
+  const { workPool } = useWorkPool(props.deployment.workPoolName ?? '')
+  const isPushWorkPool = computed(() => workPool.value?.isPushPool)
```
Comment (pleek91) [inline src/components/DeploymentDetails.vue:111]: Nit: we really don't need a computed for this. `workPool.isPushPool` can be used directly

File: src/components/FlowRunWorkQueue.vue [hunk 2]
```diff
@@ -18,6 +19,8 @@
   }>()
 
   const isNotTerminal = computed(() => props.flowRunState && !isTerminalStateType(props.flowRunState))
+  const { workPool } = useWorkPool(props.workPoolName ?? '')
+  const isPushWorkPool = computed(() => workPool.value?.isPushPool)
```
Comment (pleek91) [inline src/components/FlowRunWorkQueue.vue:23]: Nit: same suggestion here to just use `workPool.isPushPool` directly.

File: src/compositions/useWorkPool.ts
Comment (pleek91) [inline src/compositions/useWorkPool.ts:1]: ❤️

**Review by pleek91 — APPROVED** [review]
I'd remove those computed properties but otherwise this looks good to me!

**Comment by zhen0** [comment]
@pleek91 - double checking the computed check is what you were expecting before I merge!

File: src/components/FlowRunWorkQueue.vue [hunk 3]
```diff
@@ -18,6 +19,7 @@
   }>()
 
   const isNotTerminal = computed(() => props.flowRunState && !isTerminalStateType(props.flowRunState))
+  const { workPool } = useWorkPool(props.workPoolName ?? '')
```
Comment (pleek91) [inline src/components/FlowRunWorkQueue.vue:22]: Looking at this closer this probably works but might not work quite as expected. You'll probably want to do

```suggestion
  const { workPoolName } = toRefs(props)
  const { workPool } = useWorkPool(workPoolName)
```

Which will retain reactivity even if the prop starts as undefined but changes.

**Review by pleek91 — APPROVED** [review]
One last suggestion but feel free to merge.

**Pull request merged**